pytensor/tensor/math.py

```diff
@@ -2753,8 +2753,10 @@ def logsumexp(x, axis=None, keepdims=False):
     tensor
 
     """
-
-    return log(sum(exp(x), axis=axis, keepdims=keepdims))
+    result = log(
+        sum(exp(x - max(x, axis=axis, keepdims=True)), axis=axis, keepdims=keepdims)
```

Comment:
it's also a bit more complicated than that to handle cases of all infinities. The graph is pretty messy, that's why we do it later only